goldstone/core/models.py

```diff
@@ -2651,3 +2651,103 @@ def _get_traits_dict(e):
+
+        verbose_name_plural = "saved searches with alerts"
+
+    def build_alert_message_template(self, hits):
+        msg_title_template = 'Alert : %s ' + self.name + \
+                             ' triggered with %d ' + str(hits) +\
+                             'hits in last %d minutes.' + \
+                             str(self.target_interval)
+        msg_body_template = 'There were %d ' + str(hits) + \
+                            'occurrences of %s ' + str(self.name) +\
+                            'in the last %d minutes.' + '\n' +\
+                            'This query was created at : %s' +\
+                            str(self.created) + 'and last updated at : %s.' +\
+                            str(self.updated)+ '\n' +\
+                            'This query last ran from : %s' +\
+                            str(self.last_start) + 'to : %s.' +\
+                            str(self.last_end)
+        msg_params_dict = {'title': msg_title_template, 'body': msg_body_template}
+        return msg_params_dict
+
+
+class Alert(models.Model):
+    """
+        Create an alert instance object. This contains the fleshed-out
+        alert message which will be passed down to the producer interface
+        to be sent out.
+
+        This does not contain msg-params for actually sending out this alert.
+        For those details, refer to class Producer.
+    """
+
+    query = models.ForeignKey(AlertSearch)
+    name = models.CharField(max_length=64, default='generic-alert',
+                            blank=False)
+    # alert assignee vs alert receiver, can be a person vs mailing list
+    owner = models.CharField(max_length=64, default='goldstone',
+                             help_text='alert assignee, individual entity')
+
+    msg_title = models.CharField(max_length=256, default='Alert notification')
+    msg_body = models.CharField(max_length=1024,
+                                default='This is an alert notification')
+    created = CreationDateTimeField(editable=False, blank=False, null=False)
+
+
+class Producer(models.Model):
+    """
+        Generic interface class for an alert producer. This contains generic
+        producer related information such as sender, receiver names, id's,
+        host, auth-params etc
+
+        Specific types like email, slack, HTTP-POST etc inherit from this
+        class with specific connection attributes
+    """
+    query = models.ForeignKey(AlertSearch, default=1)
+
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def send(self, alert):
+        raise NotImplementedError("Producer must implement send.")
+
+
+class EmailProducer(Producer):
+    """
+        Specific interface class to prepare and send an email.
+        Class gets all of its email contents from the parent producer class.
+        This class only contains methods specific to a mailing interface.
+    """
+
+    sender = models.CharField(max_length=64, default='goldstone-bot@solinea.com')
```

Comment:
A solinea email address is not appropriate here.  There is an EXTERNAL_HOSTNAME environment variable that we can map to a Django setting like:

`EMAIL_ALERT_DEFAULT_SENDER = 'no-reply@%s' % os.environ.get('EXTERNAL_HOSTNAME', '')`

and use that as the default sender.